models/FreeLoan.v1.json

```diff
@@ -11,17 +11,22 @@
       ]
     },
     "yearly_rate": {
-      "description": "Taux d'intérêt annuel en %. Mettre 2.1 pour 2.1% par exemple.",
-      "maximum": 100,
-      "minimum": 0,
-      "type": "number"
-    },
-    "yearly_rates": {
-      "description": "Taux d'intérêt annuels en %, dans le cadre d'un prêt à taux variable. Mettre 2.1 pour 2.1% par exemple.",
-      "type": "array",
-      "items": {
-        "$ref": "./RatePeriod.v1.json"
-      }
+      "oneOf": [
```

Comment:
What does it mean for clients using this?
Can't we ALWAYS make it an array? And for the nominal case have only one field in the array instead of having to "switch" depending on the type? Not even sure how it would work in typescript for example, have you tried the generator?

Reply:
It's not very friendly to fill an object or even worse, an array, in case the rate is fixed, no?
Regarding if it will work, indeed, it looked obvious it will, to me, but maybe not.

Let me generate